### ISSUE TITLE
I need a detailed example for Context Menu in different Components

### ISSUE THREAD

**Issue by labs-scnm** [isaacplmann/angular2-contextmenu#32]
Hi,

is it possible to detailed the Example https://github.com/isaacplmann/angular2-contextmenu#context-menu-in-a-different-component ?

How can I connect the `<context-menu>` from the parent component on the child component with the `ContextMenuService`?

Thanks in advance for any help or tip!

BR
Sebastian

**Comment by isaacplmann** [comment]
The [two snippets of code](https://github.com/isaacplmann/angular2-contextmenu#context-menu-in-a-different-component) are both in the child component (or wherever you are right-clicking to show the context menu).  The parent component just needs to include this in the template:

```html
<context-menu></context-menu>
```

There could be some issues if the triggering component is in a lazy-loaded module and the context-menu element is in a different module, but that doesn't sound like your scenario.  If you're still having trouble, I can work up a plunker.

**Comment by labs-scnm** [comment]
That's like my first try, but it dosen't work.

Here my simplified example, of what I tried:

`parent.ts`
```javascript
import {IItem} from './item.interface';
import {ContextMenuComponent} from 'angular2-contextmenu';

@Component({
  ...
})
export class ParentComponent {
  public items: IItem[] = [
      { name: 'John', otherProperty: 'Foo' },
      { name: 'Joe', otherProperty: 'Bar' }
  ];

  @ViewChild('basicMenu') public basicMenu: ContextMenuComponent;

  ...
}
```
`parent.html`
```html
<app-child *ngFor="let item of items" [item]="item"></app-child>
<context-menu></context-menu>
```

`child.ts`
```javascript
import {IItem} from './item.interface';
import { ContextMenuService } from 'angular2-contextmenu';

@Component({
  ...
})
export class ChildClass {
  @Input() item: IItem[];

  constructor(private contextMenuService: ContextMenuService) {}

  public onContextMenu($event: MouseEvent, item: any): void {
    this.contextMenuService.show.next({
      event: $event,
      item: item,
    });
    $event.preventDefault();
    $event.stopPropagation();
  }

  ...
}
```
`child.html`
```html
<article (contextmenu)="onContextMenu($event, item)"></article>
```

**Comment by isaacplmann** [comment]
I started writing up a sample when I realised what the issue probably is.  You need to define some context menu items in your context menu.

In the parent template:

```html
<context-menu>
  <template contextMenuItem (execute)="showMessage('Hi, ' + $event.item.name)">
    Say hi!
  </template>
  <template contextMenuItem divider="true"></template>
  <template contextMenuItem let-item (execute)="showMessage($event.item.name + ' said: ' + $event.item.otherProperty)">
    Bye, {{item?.name}}
  </template>
</context-menu>
```

Putting `<context-menu></context-menu>` with nothing inside will show a blank context menu - which you wouldn't be able to see.

**Comment by labs-scnm** [comment]
Sorry, this was my fail, of course there are some items on it.

*I updated my example.

**Comment by isaacplmann** [comment]
I got parent-child contextmenus working locally, but am having trouble getting plunker to import the angular2-contextmenu library correctly.  I don't see anything wrong with your code.

Make sure `onContextMenu` is actually being called.  Put a console.log in there to see that there is an event and an item.

**Comment by labs-scnm** [comment]
Ok, thanks a lot for your advise!
I will try it tomorrow from scratch again. 😉

**Comment by isaacplmann** [comment]
Any progress solving this issue?

**Comment by kotrao** [comment]
Hi,

If I have 2 context menus in the parent component, how does the child communicate to parent about which context menu to show?

Thanks,
Rao

**Comment by isaacplmann** [comment]
The child needs to have a reference to the `ContextMenuComponent` that it is trying to open.  It can then call:

```ts
this.contextMenuService.show.next({
    contextMenu: this.contextMenu,
    event: $event,
    item: item,
});
```

There are a number of ways to pass the reference down, for example:

Parent component:
```html
<child contextMenu="contextMenu2"></child>
<context-menu #contextMenu1>...</context-menu>
<context-menu #contextMenu2>...</context-menu>
```

Child component:
```ts
@Input() contextMenu: ContextMenuComponent;
```

Note: If the `ContextMenuService.show.next` is called without a `contextMenu` property, all `ContextMenuComponent`s will be triggered.  I should document this better.

**Comment by kotrao** [comment]
Thanks for your response. I have tried it, but it is showing all the menus still. I also checked onContextMenu method in child just before calling ContextMenuService.show.next to see if it is actually receiving the contextMenu instance from the parent. It does receive the instance, but it is not working as you explained. 

I looked at the method onMenuEvent in ContextMenuComponent code on github. The only reference to the contextMenu being passed is as follows. After that, I am not sure how it is using the passed in contextMenu to display only that.


    let { actions, contextMenu, event, item } = menuEvent;
    if (contextMenu && contextMenu !== this) {
      this.hideMenu();
      return;
    }
